app/components/Edit/EditServices.js

```diff
@@ -1,206 +1,97 @@
 import React, { Component } from 'react';
-import Loader from '../Loader';
-import EditNotes from './EditNotes';
-import EditSchedule from './EditSchedule';
+import PropTypes from 'prop-types';
 import { createTemplateSchedule } from '../../utils/index';
-import MultiSelectDropdown from './MultiSelectDropdown';
+import ProvidedService from './ProvidedService';
 
 class EditServices extends Component {
   constructor(props) {
     super(props);
 
+    const { services } = props;
+    const existingServices = services ?
+      services.map((service) => {
+        service.key = service.id;
+        console.log(service);
+        return service;
+      }) : [];
+
     this.state = {
       services: {},
-      existingServices: props.services ? props.services.map((service) => {
-        let newService = service;
-        newService.key = service.id;
-        return newService;
-      }) : [],
+      existingServices,
       uuid: -1
     };
 
-    this.renderServices = this.renderServices.bind(this);
     this.handleServiceChange = this.handleServiceChange.bind(this);
     this.addService = this.addService.bind(this);
   }
 
+  /* @method handleServiceChange
+   * @description Updates the service with any changes made
+   * @param {string} key a unique identifier to find a service
+   * @param {object} service the service to be updated
+   * @returns {void}
+   */
   handleServiceChange(key, service) {
-    let services = this.state.services;
+    const services = this.state.services;
     services[key] = service;
     this.setState({
-      services: services
-    }, function() {
+      services
+    }, () => {
       this.props.handleServiceChange(this.state);
     });
   }
 
+  /* @method addService
+   * @description Creates a brand new service
+   */
   addService() {
-    let existingServices = this.state.existingServices;
-    let newUUID = this.state.uuid - 1;
+    const { existingServices } = this.state,
+      newUUID = this.state.uuid - 1;
```

Comment:
I know you didn't write this line originally, but this is definitely not a safe way of generating a new UUID.

Reply:
Hahaha, yeah; for sure.